test/torch_test.py

```diff
@@ -985,6 +985,30 @@ def test_mpc_mul(self):
         self.mpc_mul(3, 5)
         self.mpc_mul(2 ** 12, 2 ** 12)
 
+    def test_mpc_matmul(self):
+        x = torch.LongTensor([[1, 2], [3, 4]])
+        y = torch.LongTensor([[5, 6], [7, 8]])
+
+        x = x.share(bob, alice)
+        y = y.share(bob, alice)
+
+        assert (x.mm(y).get() - torch.LongTensor([[18, 22], [43, 49]])).abs().sum() < 5
+
+    def test_mpc_negation_and_subtraction(self):
+
+        x = torch.LongTensor([[1, 2], [3, 4]])
+        y = torch.LongTensor([[5, 6], [7, 8]])
+
+        x = x.share(bob, alice)
+        y = y.share(bob, alice)
+
+        z = -x
```

Comment:
Same here, I would mix signs if possible

Reply:
cool - fixing.. one sec.

Reply:
crap... good call... this part is broken

Reply:
if they're all negative or all positive it works great, but not if it's a mix